borsh/src/de/mod.rs

```diff
@@ -144,6 +145,92 @@ impl BorshDeserialize for usize {
     }
 }
 
+mod non_zero {
+    use super::*;
+    use core::num::*;
+
+    trait NonZero {
+        type Primitive;
+    }
```

Comment:
This trait doesn't seem necessary, since you're manually including the internal type in the macro, you can just generate the impl without it. I think this trait just adds complexity imo